packages/contracts/src/PotLuck.sol

```diff
@@ -0,0 +1,199 @@
+    mapping(bytes32 => bool) public hasWon; // keccak256(potId,user) -> bool;
+
+    event PotCreated(uint256 indexed potId, address indexed creator);
+    event Joined(uint256 indexed potId, uint32 roundId,address indexed user);
+    event Payout(uint256 indexed potId, address indexed winner, uint256 amount, uint32 round);
+
+    constructor(uint256 _platformFeePct) Ownable(msg.sender) {
+        require(_platformFeePct <= 100, "Fee pct <= 100");
+        platformFeePct = _platformFeePct;
+    }
+
+    /**
+     * @notice Create a new public pot with specified parameters.
+     * @param token Address of the ERC20 token contract.
+     * @param entryAmount USDC amount each participant must contribute.
+     * @param periodSeconds Duration of each round in seconds.
+     */
+    function createPotLuck(
+        address token,
+        uint256 entryAmount,
+        uint256 periodSeconds
+    ) external {
+        require(entryAmount > 0, "Entry > 0");
+        require(periodSeconds >= 1 hours, "Period must be >=1h");
+
+        uint256 currentPotCount = potCount++;
+        Pot storage p = pots[currentPotCount];
+        p.id = currentPotCount;
+        p.token = token;
+        p.entryAmount = entryAmount;
+        p.period = periodSeconds;
+        p.isPublic = true;
+        p.totalParticipants = 1;
+        p.deadline = block.timestamp + periodSeconds;
+        p.participants.push(msg.sender);
+
+        IERC20(token).safeTransferFrom(msg.sender, address(this), entryAmount);
+        p.balance = entryAmount;
+
+        bytes32 joinKey = keccak256(abi.encodePacked(currentPotCount, uint32(0), msg.sender));
+        hasJoinedRound[joinKey] = true;
+
+        emit PotCreated(currentPotCount, msg.sender);
+        emit Joined(currentPotCount, 0,msg.sender);
+    }
+
+    /**
+     * @notice Join an existing pot by depositing entry amount.
+     * @param potId Identifier of the pot.
+     */
+    function joinPot(uint256 potId) external {
+        Pot storage p = pots[potId];
+        require(p.balance > 0, "Pot not exist");
+        require(block.timestamp < p.deadline, "Round ended");
+        require(p.totalParticipants < MAX_PARTICIPANTS, "Full");
+        bytes32 joinKey = keccak256(abi.encodePacked(potId, p.round, msg.sender));
+        require(!hasJoinedRound[joinKey], "Already joined this round");
+
+        if(p.round ==0){
+            p.totalParticipants++; 
+        }
+
+
+        IERC20(p.token).safeTransferFrom(msg.sender, address(this), p.entryAmount);
+        p.balance += p.entryAmount;
+        p.participants.push(msg.sender);
+        
+        hasJoinedRound[joinKey] = true;
+
+        emit Joined(potId,p.round, msg.sender);
+    }
+
+    /**
+     * @notice Trigger payout for pot after deadline; picks a random winner, sends pot minus fee.
+     *         Resets round for next period.
+     * @param potId Identifier of the pot.
+     */
+    function triggerPayout(uint256 potId) external {
```

Comment:
I think this is missing logic for the case where the round is over but not all participants have paid.